src/notifications/tests/test_models.py

```diff
@@ -63,3 +65,90 @@ def new_notification(sender, **kwargs):
 
         notes_unread = Notification.objects.all_unread(self.student)
         self.assertEqual(notes_unread.count(), 1)
+
+    def test_notification_str(self):
+        """ 
+            Check if notification string method leaves behind un-stripped html tags except <img>.
+        """ 
+        # CONSTANTS
+        User = get_user_model()
+        
+        admin = User.objects.get(username="admin")
+        student = baker.make(User, username="student")
+        quest = baker.make("quest_manager.Quest", name="Quest")
+        quest_submission = baker.make("quest_manager.QuestSubmission", user=student, quest=quest)
+
+        # content type
+        comment_content_type = ContentType.objects.get(app_label="comments", model="comment")
+        quest_submission_content_type = ContentType.objects.get(app_label="quest_manager", model="questsubmission")
+
+        def create_notification(text="Sample Text"):
+            """ 
+                Quick helper function that spits out different notifications depending on text var
+            """
+            comment = baker.make("comments.Comment", text=text)
+            notification = Notification.objects.create(
+                recipient=student,
+                sender_object=admin,
+                verb="",
+
+                action_object=comment,
+                action_object_id=comment.pk,
+                action_content_type=comment_content_type,
+
+                target_object=quest_submission,
+                target_object_id=quest_submission.pk,
+                target_content_type=quest_submission_content_type,
+
+                font_icon="<i class='fa fa-info-circle'></i>",
+            )
+            return notification
+
+        def notification_start_text(notification_id):
+            """  
+                Recreates string tag for testing purposes. Equivalent to: str(notification) up until action_object
+
+                '{notification_start_text(id)}{action_object}" </a>' <=> str(notification)
+            """
+            notification_url = reverse('notifications:read', args=[notification_id])
+            submission_url = reverse('quest_manager:submission', args=[quest_submission.pk])
+            chars = f"{admin.username}  <a href='https://tenant.test.com{notification_url}?next={submission_url}'> <em>{quest.name}</em> with \""
+            return chars
+
+        notification_end_text = '"</a>'
+
+        def notification_test_helper(test_case, expected_case):
+            """ 
+                Helper for tests. Just here to prevent extra lines of boilerplate code
+            """ 
+            notification = create_notification(text=test_case)
+            strip_count = len(notification_start_text(notification.pk))
+
+            # this is just to get the text without the enclosing <a> tag
+            notification_text = str(notification)[strip_count:-len(notification_end_text)]
+
+            self.assertEqual(notification_text, expected_case)
```

Comment:
Put this assert statement in each individual test.  That way when a test fails, we know exactly what's causing it.

Reply:
done